xbmc/guilib/GUIControlFactory.cpp

```diff
@@ -271,12 +272,14 @@ bool CGUIControlFactory::GetDimensions(const TiXmlNode *node, const char *leftTa
       { // centre given, so fill to edge of parent
         width = std::max(0.0f, std::min(parentSize - center, center) * 2);
         left = center - width/2;
-        hasLeft = hasWidth = true;
+        hasLeft = true;
+        hasWidth = true;
       }
     }
     else if (hasLeft) // neither right nor center specified
     {
       width = std::max(0.0f, parentSize - left); // if left=0, this fills to parent
+      hasWidth = true;
```

Comment:
Maybe im not following this completely, but these changes seem to just set Haswidth/hasleft true for every codepath. Therefore the return will always just be true?

Couldnt just the return be changed to true, and leave everything as is?

Reply:
To add to that, wouldnt this change effectively make this block obsolete?

https://github.com/xbmc/xbmc/blob/8ababf52c9604419d12dad8f214ec2b8f578be3e/xbmc/guilib/GUIControlFactory.cpp#L785-L802

I havent looked into the history of any of that, so dont have any further context around its use.

Reply:
> Couldnt just the return be changed to true, and leave everything as is?

From a propositional standpoint, there's no fall-through else, so valid code paths can have false be returned. I don't see how to make anything obsolete without changing the logic.